ql/CMakeLists.txt

```diff
@@ -4,11 +4,14 @@ get_quantlib_library_name(QL_OUTPUT_NAME)
 
 file(GLOB_RECURSE QUANTLIB_FILES "*.hpp" "*.cpp")
 if(WIN32)
-    # Windows needs to link static library (nothing is declared to export in QuantLib)
-    add_library(${QL_OUTPUT_NAME} STATIC ${QUANTLIB_FILES})
+    if (${BUILD_SHARED_LIBS})
+        # Windows needs to link static library (nothing is declared to export in QuantLib)
+        message(WARNING "You should build QuantLib in STATIC mode as nothing is exported from the library. Use BUILD_SHARED_LIBS=OFF")
```

Comment:
@lballabio here, just a warning (it won't be able to link examples and tests) or an error?

Reply:
If the default is to use static and we get here because the user choose shared explicitly, I'd go with an error.